core/src/test/java/kafka/server/SharePartitionManagerTest.java

```diff
@@ -1128,4 +1141,156 @@ public void testShareFetchContextResponseSize() {
+        FetchParams fetchParams = new FetchParams(ApiKeys.SHARE_FETCH.latestVersion(), FetchRequest.FUTURE_LOCAL_REPLICA_ID, -1, 0,
+                1, 1024 * 1024, FetchIsolation.HIGH_WATERMARK, Optional.empty());
+        Uuid fooId = Uuid.randomUuid();
+        Uuid barId = Uuid.randomUuid();
+        TopicIdPartition tp0 = new TopicIdPartition(fooId, new TopicPartition("foo", 0));
+        TopicIdPartition tp1 = new TopicIdPartition(fooId, new TopicPartition("foo", 1));
+        TopicIdPartition tp2 = new TopicIdPartition(barId, new TopicPartition("bar", 0));
+        TopicIdPartition tp3 = new TopicIdPartition(barId, new TopicPartition("bar", 1));
+        TopicIdPartition tp4 = new TopicIdPartition(fooId, new TopicPartition("foo", 2));
+        TopicIdPartition tp5 = new TopicIdPartition(barId, new TopicPartition("bar", 2));
+        TopicIdPartition tp6 = new TopicIdPartition(fooId, new TopicPartition("foo", 3));
+        Map<TopicIdPartition, Integer> partitionMaxBytes = new HashMap<>();
+        partitionMaxBytes.put(tp0, 40000);
+        partitionMaxBytes.put(tp1, 40000);
+        partitionMaxBytes.put(tp2, 40000);
+        partitionMaxBytes.put(tp3, 40000);
+        partitionMaxBytes.put(tp4, 40000);
+        partitionMaxBytes.put(tp5, 40000);
+        partitionMaxBytes.put(tp6, 40000);
+
+        ReplicaManager replicaManager = Mockito.mock(ReplicaManager.class);
+        SharePartitionManager sharePartitionManager = new SharePartitionManager(replicaManager, new MockTime(),
+                new SharePartitionManager.ShareSessionCache(10, 1000));
+
+        doAnswer(invocation -> {
+            sharePartitionManager.releaseFetchQueueAndPartitionsLock(groupId, partitionMaxBytes.keySet());
+            return null;
+        }).when(replicaManager).fetchMessages(any(), any(), any(ReplicaQuota.class), any());
+
+        sharePartitionManager.fetchMessages(groupId, memberId1.toString(), fetchParams, Arrays.asList(tp0, tp1, tp2, tp3), partitionMaxBytes);
+        Mockito.verify(replicaManager, times(1)).fetchMessages(
+                any(), any(), any(ReplicaQuota.class), any());
+
+        sharePartitionManager.fetchMessages(groupId, memberId1.toString(), fetchParams, Collections.singletonList(tp4), partitionMaxBytes);
+        Mockito.verify(replicaManager, times(2)).fetchMessages(
+                any(), any(), any(ReplicaQuota.class), any());
+
+        sharePartitionManager.fetchMessages(groupId, memberId1.toString(), fetchParams, Arrays.asList(tp5, tp6), partitionMaxBytes);
+        Mockito.verify(replicaManager, times(3)).fetchMessages(
+                any(), any(), any(ReplicaQuota.class), any());
+    }
+
+    @Test
+    public void testMultipleConcurrentShareFetches() throws InterruptedException {
+
+        String groupId = "grp";
+        Uuid memberId1 = Uuid.randomUuid();
+        FetchParams fetchParams = new FetchParams(ApiKeys.SHARE_FETCH.latestVersion(), FetchRequest.FUTURE_LOCAL_REPLICA_ID, -1, 0,
```

Comment:
I'm surprised by FUTURE_LOCAL_REPLICA_ID here. Isn't this a normal consumer really?

Reply:
I am just using `fetchParams` as a dummy value here. The function `replicaManager.fetchMessages()` is mocked, so it should not affect anything. Please let me know what should be the value used here, because in [KafkaApis.scala](https://github.com/confluentinc/kafka/blob/kip-932/core/src/main/scala/kafka/server/KafkaApis.scala#L1429), I see `FUTURE_LOCAL_REPLICA_ID` being used in the actual share fetch request handling.

Reply:
`ORDINARY_CONSUMER_ID`? That's my recommendation.

Reply:
sure, I'll make the change in my test